fiwareskuld/expired_users.py

```diff
@@ -23,150 +23,197 @@
-        maintained in the internal attribute "token" of the class.
-        :return: The admin token to be used in the X-Auth-Token header
+        k = self.keystoneclient
+        role = k.roles.find(name="trial")
+        return set(e.user['id'] for e in k.role_assignments.list(
+            role=role.id))
+
+    def get_community_user_ids(self):
+        """Get a set of community users; only the ids
+        :return: a set of user ids, corresponding to trial users.
         """
+        k = self.keystoneclient
+        role = k.roles.find(name="community")
+        return set(e.user['id'] for e in k.role_assignments.list(
+            role=role.id))
+
+    def get_basic_users_ids(self):
+        """Get a set of basic users; only the ids
+        :return: a set of user ids, corresponding to trial users.
+        """
+        k = self.keystoneclient
+        role = k.roles.find(name="basic")
+        return set(e.user['id'] for e in k.role_assignments.list(
+            role=role.id))
+
+    def get_basic_users(self):
+        """Get the list of basic users; the full objects are included.
+        :return: a list of trial users
+        """
+        user_ids = self.get_basic_users_ids()
+        return list(user for user in self.keystoneclient.users.list() if user.id in user_ids)
 
-        self.__check_credentials()
-
-        payload = "{\"auth\":{\"tenantName\":\"%s\"," \
-                  "\"passwordCredentials\":{\"username\":\"%s\",\"password\":\"%s\"}}}" \
-                  % (self.__tenant, self.__username, self.__password)
-        headers = {'content-type': 'application/json'}
-        url = self.KEYSTONE_ENDPOINT + self.v20 + "tokens"
-        r = requests.post(url=url, data=payload, headers=headers)
-
-        rjson = json.loads(r.text)
-
-        if r.status_code == 200:
+    def get_trial_users(self):
+        """Get the list of trial users; the full objects are included.
+        :return: a list of trial users
+        """
+        user_ids = self.get_trial_user_ids()
+        return list(user for user in self.keystoneclient.users.list() if user.id in user_ids)
 
-            self.token = rjson['access']['token']['id']
+    def get_community_users(self):
+        """Get the list of community users; the full objects are included.
+        :return: a list of community users
+        """
+        user_ids = self.get_community_user_ids()
 
-            logger.info("Admin token requested: %s", self.token)
-        else:
-            raise Exception(rjson['error']['message'])
+        d = list(user for user in self.keystoneclient.users.list() if user.id in user_ids)
+        return d
 
-        return self.token
+    def get_users(self):
+        """Get the list of trial users; the full objects are included.
+        :return: a list of trial users
+        """
+        return self.keystoneclient.users.list()
 
-    def get_list_trial_users(self):
+    def delete_trial_users(self):
```

Comment:
delete functionalities is something that it is created outside of this component. The purpose of expired_users.py was to get the lists of users but not delete them...